src/vm_interp.rs

```diff
@@ -117,44 +117,62 @@ pub fn execute_vm_interp(syscall_context: SyscallContext) -> Option<SyscallEffec
         _ => SBPFVersion::V0,
     };
 
-    // stub syscalls
-    let syscall_reg = unstubbed_runtime.get_function_registry(sbpf_version);
-    let mut stubbed_syscall_reg = FunctionRegistry::<BuiltinFunction<TestContextObject>>::default();
-
-    for (key, (name, _)) in syscall_reg.iter() {
-        stubbed_syscall_reg
-            .register_function(key, name, SyscallStub::vm)
-            .unwrap();
-    }
-
+    let enable_stack_frame_gaps = (sbpf_version == SBPFVersion::V0)
+        || !feature_set.is_active(&bpf_account_data_direct_mapping::id());
```

Comment:
Doesn't this mean that we will never test SBPF V0 programs with stack frame gaps disabled?

When direct mapping is activated everywhere, on real networks most programs will be SBPF V0, and will be executed with stack frame gaps disabled.

Agave uses `enable_stack_frame_gaps: !feature_set.is_active(&bpf_account_data_direct_mapping::id())` - what's the reason for the different condition here?

Reply:
in V1+ gaps must be disabled (the stack is reversed).
We can test V0 with and without gaps, but starting from V1 gaps must be disabled.
Agave is prob assuming that direct mapping will be enabled before V1?

Reply:
update: the expression was incorrect, it should be `&&` not `||`.
gaps are disabled in v>v0 OR dm enabled
so gaps are enabled if v==v0 && dm is disabled